packages/riipen-ui/src/components/Badge.test.jsx

```diff
@@ -0,0 +1,306 @@
+          .childAt(0)
+          .hasClass("bottom-left-rectangle")
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid anchor", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const anchor = { horizontal: "side", vertical: "up" };
+
+      mount(<Badge anchor={anchor} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("children prop", () => {
+    it("displays given children", () => {
+      const child = <div>child</div>;
+
+      const wrapper = mount(<Badge>{child}</Badge>);
+
+      expect(wrapper.containsMatchingElement(child)).toBeTruthy();
+    });
+  });
+
+  describe("classes prop", () => {
+    it("sets custom class", () => {
+      const classVariant = ["newClass"];
+
+      const wrapper = mount(<Badge classes={classVariant} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(classVariant[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("color prop", () => {
+    it("sets color class name with a valid color", () => {
+      const color = "muted";
+
+      const wrapper = mount(<Badge color={color} />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(color)
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Badge color="blue" />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("component prop", () => {
+    it("sets component with valid custom component", () => {
+      const component = "a";
+
+      const wrapper = mount(<Badge component={component} />);
+
+      expect(wrapper.find(component)).toHaveLength(1);
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("gives an error when given an invalid component", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcdefg";
+
+      mount(<Badge component={component} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("content prop", () => {
+    it("badge does not render when content is null and showZero is false", () => {
+      const wrapper = mount(<Badge />);
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("hidden")
+      ).toEqual(true);
+      expect(wrapper.text()).toEqual("");
+    });
+
+    it("badge renders with content and showZero props", () => {
+      const content = 2;
+      const showZero = true;
+
+      const wrapper = mount(<Badge content={content} showZero={showZero} />);
+
+      expect(Number(wrapper.find("Badge").text())).toEqual(2);
+    });
+  });
```

Comment:
I think we should add one more test here. "It should render truthy content" where content is just set to 1 or 2 or something. You covered the other two edge cases very well, but I think we need this base case as well.